### PR TITLE
Fix #770: Added Support for CronJob

### LINKED ISSUE #770: Is this api supports Kubernetes Cron Job?
Is this api supports Kubernetes Cron Job?
https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#schedule
If so, someone please share a sample with me.

I want to create Cron Jobs via Java API and get the statistics of it

### PR DESCRIPTION
**Pull request by rohanKanojia**
#770

### CODE REVIEW

**Comment by piyush1594** [comment]
@rohanKanojia You need to add CronJob handler and changelog

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java [hunk 1]
```diff
@@ -15,12 +15,12 @@
  */
 package io.fabric8.kubernetes.client;
 
-import io.fabric8.kubernetes.api.model.batch.DoneableJob;
-import io.fabric8.kubernetes.api.model.batch.Job;
-import io.fabric8.kubernetes.api.model.batch.JobList;
+import io.fabric8.kubernetes.api.model.batch.*;
```
Comment (piyush1594) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java]: We should ignore this practice, while doing the 3.9 upgrade, I tried to remove it as much as possible, this creates lots of problems when doing the upgrade, let's ignore * and import only which is required.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/BatchAPIGroupDSL.java [hunk 2]
```diff
@@ -15,12 +15,12 @@
  */
 package io.fabric8.kubernetes.client.dsl;
 
-import io.fabric8.kubernetes.api.model.batch.DoneableJob;
-import io.fabric8.kubernetes.api.model.batch.Job;
-import io.fabric8.kubernetes.api.model.batch.JobList;
+import io.fabric8.kubernetes.api.model.batch.*;
```
Comment (piyush1594) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/BatchAPIGroupDSL.java]: Same here as above

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/CronJobOperationsImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.dsl.internal;
+
+import io.fabric8.kubernetes.api.model.batch.*;
```
Comment (piyush1594) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/CronJobOperationsImpl.java]: Here also

File: kubernetes-client/src/main/resources/META-INF/services/io.fabric8.kubernetes.client.ResourceHandler [hunk 4]
```diff
@@ -25,6 +25,7 @@ io.fabric8.kubernetes.client.handlers.IngressHandler
 io.fabric8.kubernetes.client.handlers.KubernetesRoleBindingHandler
 io.fabric8.kubernetes.client.handlers.KubernetesRoleHandler
 io.fabric8.kubernetes.client.handlers.JobHandler
+io.fabric8.kubernetes.client.handlers.CronJobHandler
```
Comment (piyush1594) [inline kubernetes-client/src/main/resources/META-INF/services/io.fabric8.kubernetes.client.ResourceHandler]: This list is in alphabetical order, lets maintain that

File: openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java [hunk 5]
```diff
@@ -68,6 +68,9 @@
 import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinition;
 import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinitionList;
 import io.fabric8.kubernetes.api.model.apiextensions.DoneableCustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.batch.CronJob;
```
Comment (piyush1594) [inline openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java]: I think this is not required.

Comment (rohanKanojia) [reply openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java]: Yes, will remove it in next update

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java (hunk 1 above)
Comment (rohanKanojia) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BatchAPIGroupClient.java]: I think it's done automatically while importing things in IntelliJ. Not sure if it's a bad practice or not.

**Comment by oscerd** [comment]
[merge]

**Pull request merged**